src/app/mytools/sidebar-accordion/sidebar-accordion.component.ts

```diff
@@ -1,24 +1,74 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnInit, OnChanges, SimpleChanges } from '@angular/core';
 import { DockstoreTool } from 'app/shared/swagger';
 import { Observable } from 'rxjs';
 import { ToolQuery } from '../../shared/tool/tool.query';
 import { OrgToolObject } from '../my-tool/my-tool.component';
+import { KeyValue } from '@angular/common';
+import { MetadataService } from '../../shared/swagger/api/metadata.service';
+
+interface GroupEntriesByRegistry {
+  groupEntryInfo: OrgToolObject<DockstoreTool>[];
+  registryTitle: string;
+}
 
 @Component({
   selector: 'app-sidebar-accordion',
   templateUrl: './sidebar-accordion.component.html',
-  styleUrls: ['./sidebar-accordion.component.scss'],
+  styleUrls: ['./sidebar-accordion.component.scss', '../../shared/styles/my-entry-sidebar.scss'],
 })
-export class SidebarAccordionComponent implements OnInit {
-  constructor(private toolQuery: ToolQuery) {}
+export class SidebarAccordionComponent implements OnInit, OnChanges {
   @Input() openOneAtATime;
   @Input() groupEntriesObject: OrgToolObject<DockstoreTool>[];
   @Input() refreshMessage;
-
   public toolId$: Observable<number>;
+  private dockerRegistryMap = [];
+  public registryToTools: Map<string, GroupEntriesByRegistry> = new Map<string, GroupEntriesByRegistry>([]);
   activeTab = 0;
 
-  ngOnInit(): void {
+  constructor(private toolQuery: ToolQuery, private metadataService: MetadataService) {
+    this.metadataService.getDockerRegistries().subscribe((map) => {
+      this.dockerRegistryMap = map;
+      this.dockerRegistryMap.forEach((registry) => {
+        // Do not create new keys for amazon and seven bridges, to be put in additional category
+        if (registry.dockerPath !== null && registry.dockerPath !== 'public.ecr.aws')
+          this.registryToTools.set(registry.dockerPath, { groupEntryInfo: [], registryTitle: registry.friendlyName.toUpperCase() });
+      });
+      // For additional tools, to be removed once amazon and seven bridges have their own groupings
+      this.registryToTools.set('additional', { groupEntryInfo: [], registryTitle: 'ADDITIONAL' });
```

Comment:
It would be good if `'additional'` were a constant so you could have the compiler enforce that you're spelling it the same way on line 57 (you are spelling it the same way now, but it's not enforced).